src/org/labkey/test/tests/SampleSetRemoteAPITest.java

```diff
@@ -181,6 +196,328 @@ public void sampleSetWithMissingValueField() throws IOException, CommandExceptio
+
+        Run runA = new Run();
+        runA.setProperties(Collections.singletonMap("RunFileField", "foo.xls"));
+        Data runAData = new Data();
+        runAData.setName("joe");
+        runAData.setPipelinePath("/");
+        runA.setDataInputs(Arrays.asList(runAData));
+        runA.setName("saveBatchMatrials A");
+        runA.setMaterialOutputs(Arrays.asList(material5, material6));       // 5 and 6 don't exist yet, they should be created as outputs
+
+        Run runB = new Run();
+        runB.setProperties(Collections.singletonMap("RunFileField", "foo.xls"));
+        Data runBData = new Data();
+        runBData.setName("billy");
+        runBData.setPipelinePath("/");
+        runB.setDataInputs(Arrays.asList(runBData));
+        runB.setName("saveBatchMaterials B");
+        runB.setMaterialInputs(Arrays.asList(material4)); // material4 should already exist
+        runB.setMaterialOutputs(Arrays.asList(material7));
+
+        batch.getRuns().add(runA);
+        batch.getRuns().add(runB);
+
+        SaveAssayBatchCommand saveAssayBatchCommand = new SaveAssayBatchCommand(SaveAssayBatchCommand.SAMPLE_DERIVATION_PROTOCOL, batch);
+        //saveAssayBatchCommand.setAssayId(Integer.parseInt(assayIdStringValue));
+        SaveAssayBatchResponse saveAssayBatchResponse;
+        saveAssayBatchResponse = saveAssayBatchCommand.execute(createDefaultConnection(true), getProjectName() + "/" + samplesFolder);
+
+        refresh();
+        // add a bodyWebPart here to make viewing the sampleset easier while debugging
+        new PortalHelper(getDriver()).addBodyWebPart("Sample Sets");
+        refresh();
+        DataRegionTable sampleSetList =  DataRegionTable.DataRegion(getDriver()).withName("SampleSet").waitFor();
+        waitAndClick(Locator.linkWithText(sampleSetName));
+        DataRegionTable materialsList =  DataRegionTable.DataRegion(getDriver()).withName("Material").waitFor();
+
+        // get the samples from the sampleset
+        SelectRowsResponse sampleSetResponse = dgen.getRowsFromServer(createDefaultConnection(true));
+
+        // at this point, the API either doesn't work or requires some hack to make it work.
+        // the materialoutputs are present, but the API does not create samples derived from the runs.
+
+    }
+
+    private AssayDesignerPage generateAssay(String subfolderName, String assayName)
+    {
+        _containerHelper.createSubfolder(getProjectName(), subfolderName);
+        navigateToFolder(getProjectName(), subfolderName);
+        goToManageAssays();
+        AssayDesignerPage designerPage = _assayHelper.createAssayAndEdit("General", assayName);
+
+        return designerPage;
+    }
+
+    /**
+     * generates data (runs) for the gpat assay used by tests in this class.
+     * @param assayName
+     * @return
+     */
+    private List<TestDataGenerator> insertAssayData(String assayName, String assaySubfolder, FieldDefinition.LookupInfo assayLookup)
```

Comment:
It doesn't look like the parameter assaySubFolder is used.

Reply:
pulling it out